### PR TITLE
evm: `debug_traceTransaction`

### PR DESCRIPTION
**Pull request by shohamc1**
Supersedes #2007

### CODE REVIEW

File: lib/ain-cpp-imports/src/bridge.rs [hunk 1]
```diff
@@ -50,5 +50,7 @@ pub mod ffi {
         fn getNumCores() -> i32;
         fn getCORSAllowedOrigin() -> String;
         fn getNumConnections() -> i32;
+        fn isDebugEnabled() -> bool;
```
Comment (prasannavl) [inline lib/ain-cpp-imports/src/bridge.rs:53]: Let's retrospectively change this to isEthDebugRPCEnabled, isEthDebugTraceEnabled, etc.

File: lib/ain-evm/src/core.rs [hunk 2]
```diff
@@ -740,6 +755,88 @@ impl EVMCoreService {
...
+        debug!(
+            "Calling EVM at block number : {:#x}, state_root : {:#x}",
+            block_number, state_root
+        );
+
+        let vicinity = Vicinity {
+            block_number,
+            origin: caller,
+            ..Default::default()
+        };
```
Comment (sieniven) [inline lib/ain-evm/src/core.rs:785]: Not setting the full vicinity state of the backend with the block context could lead to incorrect state execution

File: lib/ain-cpp-imports/src/bridge.rs (hunk 1 above)
Comment (sieniven) [reply lib/ain-cpp-imports/src/bridge.rs:53]: Resolved on PR https://github.com/DeFiCh/ain/pull/2641

File: lib/ain-evm/src/core.rs (hunk 2 above)
Comment (sieniven) [reply lib/ain-evm/src/core.rs:785]: Resolved on PR https://github.com/DeFiCh/ain/pull/2641

**Pull request merged**